main/src/modules/contextualLayer/planner/include/planner.h

```diff
@@ -0,0 +1,80 @@
+#include "wrdac/clients/icubClient.h"
+
+using namespace std;
+using namespace yarp::os;
+using namespace wysiwyd::wrdac;
+
+class Planner : public RFModule {
+private:
+
+    ICubClient *iCub;
+    double      period;
+    Port        rpc;
+    Port        portToBehavior;
+    Port        toHomeo;
+    Port        getState;
+    BufferedPort<Bottle>        port_behavior_context;
+    Bottle avaiPlansList;
+    vector<Bottle> newPlan;
+    vector<string> plan_list;
+    vector<string> action_list;
+    vector<string> object_list;
+    vector<string> type_list;
+    vector<int> priority_list;
+    vector<int> actionPos_list;
+    vector<string>::iterator actPt;
+    vector<int>::iterator prioPt;
+    Bottle grpPlans;
+    Bottle gandalf;
+    Bottle adhd;
+    int id;
+    int fulfill;
+    int attemptCnt;
+    // Bottle* current_goal;
+    // string current_goal;
+    string objectType;
+    string object;
+
+    // bool bShouldListen;
+
+    std::string      grammarToString(std::string sPath);
+    std::string      MainGrammar;
+
+    yarp::os::Mutex mutex;
+    bool following;
+    bool ordering;
+    Port behaviorsRPC;
+    void run(Bottle args=Bottle());
+
+
+public:
+    bool configure(yarp::os::ResourceFinder &rf);
+
+    bool configurePlans(yarp::os::ResourceFinder &rf);
+
+    // bool interruptModule()
```

Comment:
Because there is another function `close()` which does that and more. Will remove it entirely.